### ISSUE TITLE
Can't upload changes to mapillary

### ISSUE THREAD

**Issue by vantu5z** [JOSM/Mapillary#27]
I can make changes (move and correct direction), see change sets, bun when push Send button, josm freeze for some time and nothing went after.
Use git version of addon.

**Comment by floscher** [comment]
Thank you for opening the issue, I'll look into it.

I can confirm this issue, when images that are already on the Mapillary servers, are modified.
Tested on Ubuntu 16.04, using OpenJDK Runtime Environment (1.8.0_111) and version [`v1.3.1-22-ge9d2e07`](https://github.com/JOSM/Mapillary/commit/e9d2e07c22a86aae545a2f4af6a80de79b32a24b) of the plugin.

**Comment by floscher** [comment]
It looks like the API rejects changesets sent to the `https://a.mapillary.com/v3/changeset` endpoint.

I sent a POST request to `https://a.mapillary.com/v3/changeset?client_id=‹client-ID›` with the headers `Content-Type: application/json` and `Authorization: Bearer ‹OAuth-token›` and the following body:
```json
{
  "change_type":"location",
  "changes":[
    {
      "image_key":"OYBIMRIKygmLilXuWMEOhw",
      "values":{
        "from":{
          "ca":186.32260131835938,
          "lat":48.74874815079043,
          "lon":8.34672063589096
        },
        "to":{
          "ca":186.32260131835938,
          "lat":48.74874462561275,
          "lon":8.346719321958886
        }
      }
    },
    {
      "image_key":"uiWpweKLXTejNezY5cTnqQ",
      "values":{
        "from":{
          "ca":108.9020004272461,
          "lat":48.7487490350641,
          "lon":8.34672063589096
        },
        "to":{
          "ca":108.9020004272461,
          "lat":48.748749453305486,
          "lon":8.34672541588523
        }
      }
    },
    {
      "image_key":"xkyvTuyz2EFMxdamuDxGGA",
      "values":{
        "from":{
          "ca":87.52902221679688,
          "lat":48.74872191,
          "lon":8.34682728
        },
        "to":{
          "ca":87.52902221679688,
          "lat":48.74875209807385,
          "lon":8.3467256994583
        }
      }
    },
    {
      "image_key":"Q5ZaPdUXabrG1FHQKVp_Yw",
      "values":{
        "from":{
          "ca":127.25493621826172,
          "lat":48.74874815079043,
          "lon":8.34672063589096
        },
        "to":{
          "ca":127.25493621826172,
          "lat":48.74874601477178,
          "lon":8.346722991907095
        }
      }
    }
  ],
  "request_comment":"JOSM-created"
}
```

The response was a `404 Not Found`.

@peterneubauer Has the API changed since the changeset feature was introduced? Or is something broken (either in the plugin or the API)?

**Comment by floscher** [comment]
The [latest release v1.3.2](https://github.com/JOSM/Mapillary/releases/tag/v1.3.2) now no longer freezes JOSM and displays error messages when submitting changesets fails.
You still can't submit changesets, but at least you get some feedback, **that** submitting didn't work, and you get a hint **why** it didn't work.
For fixing the rest; I'll need help by the @mapillary team.

**Comment by peterneubauer** [comment]
Yup,
I hear you. we are right now writing the v3 REST API for changesets and I will update the code when it's working.

**Comment by peterneubauer** [comment]
I will build the plugin and try to test upload shortly.

**Comment by peterneubauer** [comment]
Sorry, no time yet to test

**Comment by jpietri** [comment]
This should be fixed with plugin version 1.5.19.
Feedback/confirmation welcome :-)

**Comment by jpietri** [comment]
Fixed with plugin version 1.5.19;
